APE14.rst

```diff
@@ -0,0 +1,450 @@
+
+.. code-block:: python
+
+    >>> wcs.low_level_wcs.pixel_to_world_values(...)
+
+Since a single Astropy object might correspond to two non-contiguous dimensions
+in the WCS (for example the first and third world dimensions), we need to
+specify the rules for the order in which Astropy objects are returned from the
+high-level ``pixel_to_world`` method, and in which order they should be given to
+the high-level ``world_to_pixel`` method. The standard order should be that
+given by considering only the first occurrence of the coordinate alias string in
+``world_axis_object_components``. For example, if
+``world_axis_object_components`` is
+
+.. code-block:: python
+
+    [('skycoord', 'ra'), ('time', 0), ('skycoord', 'dec')]
+
+Then the order of the Astropy objects should be ``SkyCoord`` then ``Time`` (we
+essentially ignore ``('skycoord', 'dec')``). This rule will always be followed
+for ``pixel_to_world``, but on the other hand provided there is no ambiguity,
+``world_to_pixel`` could be more forgiving if the coordinates are specified in
+the wrong order (though an error should be raised if there are any ambiguities
+and the order is not the standard one).
+
+Branches and pull requests
+--------------------------
+
+N/A
+
+Implementation
+--------------
+
+The low-level API could be implemented specifically for FITS-WCS into the
+Astropy core package, along with the more general high-level Astropy object.
+Other projects can then choose to implement objects conforming to the low-level
+API in their own packages.
+
+Backward compatibility
+----------------------
+
+N/A
+
+Alternatives
+------------
+
+A possible alternative to consider is simply leave things status quo and have no
+agreed-on API. Rather instead have Astropy endorse a specific implementation
+like gwcs as the API to assume for when wcs objects are needed (e.g. NDData and
+spectroscopic objects). However, this would likely lead to less uptake of the
+upstream objects that require wcs. E.g., while ``astropy.wcs`` is a
+commonly-used case in the present, it does not support new use cases like LSST
+or the distortion models for JWST, and is tied specifically to the FITS format.
+Therefore tools that wish to support both FITS WCS *and* newer systems would
+have to write their own complex logic for doing so, as well as potentially even
+more complex logic for converting the wcs outputs into composite Python objects.
+The structure outlined in this API would make that unnecessary by instead having
+a single interface that user code can write against, and only needs to consider
+other details when creating or modifying wcs.
+
+On a more specific note, the primary reason for using a string as the key for
+the dictionary for ``world_axis_object_classes`` (and the corresponding names in
+world_axis_object_components) is because there might be multiple world axes that
+need to use the same class. Otherwise a simpler solution would have been to use
+the class object *itself* as the key.
```

Comment:
What is the problem if you use the class object itself in the case that multiple world axes use the same class?